src/isl/util.rs

```diff
@@ -446,10 +550,43 @@ impl RangeBoundaryValue {
 
         match value.ion_type() {
             IonType::Symbol => {
+                use TimestampPrecisionValue::*;
                 let sym = try_to!(try_to!(value.as_sym()).text());
                 match sym {
                     "min" => Ok(RangeBoundaryValue::Min),
                     "max" => Ok(RangeBoundaryValue::Max),
+                    "year" => Ok(RangeBoundaryValue::Value(
+                        RangeBoundaryValueType::TimestampPrecision(Year as i64),
```

Comment:
Having named `TimestampPrecisionValue` variants feels like a nice feature; why are we discarding the name and using the enum variant index? (And do we want that index to be signed?)